### PR TITLE
Neilj/1.0 upgrade notes

### PR DESCRIPTION
**Pull request by neilisfragile**
upgrade and install notes ahead of 1.0

### CODE REVIEW

File: INSTALL.md [hunk 1]
```diff
@@ -397,6 +397,27 @@ To configure Synapse to expose an HTTPS port, you will need to edit
 For those of you upgrading your TLS certificate in readiness for Synapse 1.0,
 please take a look at [our guide](docs/MSC1711_certificates_FAQ.md#configuring-certificates-for-compatibility-with-synapse-100).
 
+## Email
+Email
```
Comment (babolivier) [inline INSTALL.md]:
```suggestion

```

File: INSTALL.md [hunk 2]
```diff
@@ -397,6 +397,27 @@ To configure Synapse to expose an HTTPS port, you will need to edit
...
+If Synapse is not configured with an SMTP server, password reset via email will
+ be disabled.
+
+Previous versions of Synapse delegated the job of sending this email to an
+identity server. If the identity server was somehow malicious or became
+compromised, it would be theoretically possible to hijack an account through
+this means.
+
+If you are absolutely certain that you wish to continue using an identity
+server for email, set ``enable_password_reset_from_is`` to ``true``.
```
Comment (babolivier) [inline INSTALL.md]:
```suggestion
server for password resets, set ``enable_password_reset_from_is`` to ``true``.
```

File: UPGRADE.rst [hunk 3]
```diff
@@ -49,6 +49,51 @@ returned by the Client-Server API:
...
+In rare cases, it may be desirable to disable certificate checking: for
+example, it might be essential to be able to federate with a given legacy
+server in a closed federation. This can be done in one of two ways:- *
+Configure the global switch ``federation_verify_certificates`` to ``false``. *
+Configure a whitelist of server domains to trust via
+``federation_certificate_verification_whitelist``.
+
+See the `sample configuration file
+<https://github.com/matrix-org/synapse/blob/master/docs/sample_config.yaml>`_
+for more details on these settings. Password reset emails When a user requests
```
Comment (babolivier) [inline UPGRADE.rst]: Not sure if `Password reset emails` is supposed to be a title or some text that got in the way here.

File: UPGRADE.rst [hunk 4]
```diff
@@ -49,6 +49,51 @@ returned by the Client-Server API:
...
+Synapse is not configured with an SMTP server, password reset via email will be
+disabled.
+
+To configure an SMTP server for Synapse, modify the configuration section
+headed ``email``, and be sure to have at least the ``smtp_host``, ``smtp_port``
+and ``notif_from`` fields filled out. You may also need to set ``smtp_user``,
+``smtp_pass``, and ``require_transport_security``..
+
+If you are absolutely certain that you wish to continue using an identity
+server for email, set ``enable_password_reset_from_is`` to ``true``.
```
Comment (babolivier) [inline UPGRADE.rst]:
```suggestion
server for password resets, set ``enable_password_reset_from_is`` to ``true``.
```

File: INSTALL.md (hunk 2 above)
Comment (anoadragon453) [inline INSTALL.md]:
```suggestion
server for email, set ``disable_local_password_reset`` to ``true``
under the ``email:`` configuration section.
```

File: INSTALL.md [hunk 5]
```diff
@@ -397,6 +397,27 @@ To configure Synapse to expose an HTTPS port, you will need to edit
 For those of you upgrading your TLS certificate in readiness for Synapse 1.0,
 please take a look at [our guide](docs/MSC1711_certificates_FAQ.md#configuring-certificates-for-compatibility-with-synapse-100).
 
+## Email
+It is desirable for Synapse to have the capability to send email, particularly
```
Comment (richvdh) [inline INSTALL.md]:
```suggestion

It is desirable for Synapse to have the capability to send email.
```

File: INSTALL.md [hunk 6]
```diff
@@ -397,6 +397,27 @@ To configure Synapse to expose an HTTPS port, you will need to edit
...
+in the case of password reset.
```
Comment (richvdh) [inline INSTALL.md]:
```suggestion
For example, this is required to support the 'password reset' feature.
```

File: INSTALL.md [hunk 7]
```diff
@@ -397,6 +397,27 @@ To configure Synapse to expose an HTTPS port, you will need to edit
...
+
+To configure an SMTP server for Synapse, modify the configuration section
+headed ``email``, and be sure to have at least the ``smtp_host``, ``smtp_port``
+and ``notif_from`` fields filled out. You may also need to set ``smtp_user``,
+``smtp_pass``, and ``require_transport_security``..
+
+If Synapse is not configured with an SMTP server, password reset via email will
+ be disabled.
+
+Previous versions of Synapse delegated the job of sending this email to an
```
Comment (richvdh) [inline INSTALL.md]: Previous to what? this paragraph is probably redundant in INSTALL.md anyway.

File: UPGRADE.rst (hunk 3 above)
Comment (neilisfragile) [reply UPGRADE.rst]: done

File: INSTALL.md [hunk 8]
```diff
@@ -397,6 +397,27 @@ To configure Synapse to expose an HTTPS port, you will need to edit
...
+identity server. If the identity server was somehow malicious or became
+compromised, it would be theoretically possible to hijack an account through
+this means.
+
+If you are absolutely certain that you wish to continue using an identity
+server for email, set ``disable_local_password_reset`` to ``true``
```
Comment (babolivier) [inline INSTALL.md]:
```suggestion
server for password resets, set ``disable_local_password_reset`` to ``true``
```

Comment (anoadragon453) [inline INSTALL.md]:
```suggestion
server for email, set ``trust_identity_server_for_password_resets`` to ``true``
```

And also explain that if this is set to `false` and the `email:` block has not been filled out, then password resets will be disabled.

File: UPGRADE.rst [hunk 9]
```diff
@@ -49,6 +49,51 @@ returned by the Client-Server API:
...
+Synapse is not configured with an SMTP server, password reset via email will be
+disabled.
+
+To configure an SMTP server for Synapse, modify the configuration section
+headed ``email``, and be sure to have at least the ``smtp_host``, ``smtp_port``
+and ``notif_from`` fields filled out. You may also need to set ``smtp_user``,
+``smtp_pass``, and ``require_transport_security``..
+
+If you are absolutely certain that you wish to continue using an identity
+server for password resets, set ``enable_password_reset_from_is`` to ``true``.
```
Comment (jryans) [inline UPGRADE.rst]: `enable_password_reset_from_is` seems to disagree with the option name mentioned in INSTALL.md for this.

File: UPGRADE.rst [hunk 10]
```diff
@@ -49,6 +49,51 @@ returned by the Client-Server API:
     # configured on port 443.
     curl -kv https://<host.name>/_matrix/client/versions 2>&1 | grep "Server:"
 
+Upgrading to v1.0
+=================
+
+Validation of TLS certificates Synapse v1.0 is the first release to enforce
+validation of TLS certificates for the federation API. It is therefore
+essential that your certificates are correctly configured. See the `FAQ
+<docs/MSC1711_certificates_FAQ.md>`_ for more information.
```
[the remark this answers is not in the document]
Comment (richvdh) [reply UPGRADE.rst]: otoh that's a thing we can fix between RC and release, so \<shrug>

File: INSTALL.md (hunk 7 above)
Comment (neilisfragile) [reply INSTALL.md]: modified to remove time reference,

File: UPGRADE.rst [hunk 11]
```diff
@@ -49,6 +49,51 @@ returned by the Client-Server API:
...
+
+Note, v1.0 installations will also no longer be able to federate with servers
+that have not correctly configured their certificates.
+
+In rare cases, it may be desirable to disable certificate checking: for
+example, it might be essential to be able to federate with a given legacy
+server in a closed federation. This can be done in one of two ways:- *
```
[the remark this answers is not in the document]
Comment (neilisfragile) [reply UPGRADE.rst]: wrap plug in fail

File: UPGRADE.rst [hunk 12]
```diff
@@ -49,6 +49,51 @@ returned by the Client-Server API:
...
+Configure the global switch ``federation_verify_certificates`` to ``false``. *
```
[the remark this answers is not in the document]
Comment (neilisfragile) [reply UPGRADE.rst]: as above

File: UPGRADE.rst [hunk 13]
```diff
@@ -49,6 +49,51 @@ returned by the Client-Server API:
...
+Configure a whitelist of server domains to trust via
```
[the remark this answers is not in the document]
Comment (neilisfragile) [reply UPGRADE.rst]: as above

File: UPGRADE.rst (hunk 9 above)
[the remark this answers is not in the document]
Comment (neilisfragile) [reply UPGRADE.rst]: trust_identity_server_for_password_resets

File: UPGRADE.rst (hunk 10 above)
[the remark this answers is not in the document]
Comment (neilisfragile) [reply UPGRADE.rst]: okay agreed - I'll ship as is and make a separate non-blocking PR to tear the FAQ apart.

**Pull request merged**